File: Lecture031 Recursion Day1/power.cpp
#include<iostream> 
using namespace std;

int power(int n) {

    //base case 
    if(n == 0)
        return 1;

    //recursive relation
    return 2 * power(n-1);    
}

int main() {

    int n;
    cin >> n;

    int ans = power(n);

    cout << ans << endl;

    return 0;
}

File: 005_Bitwise_Operators,_For_Loops,_Operator_Precedence_and_Variable_Scoping/Power.java
public class Power {
    // calculate Power of a Number
    public static void main(String [] args){
        int base = 3;
        int power = 2;
        int ans = 1;

        while(power > 0){
            if((power & 1) == 1){
                ans *= base;
            }
            power = power >> 1;
            base *= base;
        }
        System.out.println(ans);
    }
}
